api/envoy/extensions/filters/network/tcp_proxy/v3/tcp_proxy.proto

```diff
@@ -68,6 +68,19 @@ message TcpProxy {
 
     // The hostname to send in the synthesized CONNECT headers to the upstream proxy.
     string hostname = 1 [(validate.rules).string = {min_len: 1}];
+
+    // Use POST method instead of CONNECT method to tunnel the TCP stream.
+    // The 'protocol: bytestream' header is also NOT set to comply with the HTTP spec.
+    // Only HTTP/2 is supported.
+    //
+    // The upstream proxy is expected to convert POST back to CONNECT.
+    bool use_post = 2;
+
+    // Additional request headers to upstream proxy. This is mainly used to
+    // trigger upstream to convert POST requests back to CONNECT requests.
+    // The hostname header can be overridden by this field.
```

Comment:
Also I don't think we should allow overriding :method  - setting hostname but overriding with :authority here seems confusing.  Ditto use_post but setting :method.
I'd disallow, add config checks, and comment.

Reply:
Should be disallowed now.